### PR TITLE
Dark mode #569

### PR DESCRIPTION
**Pull request by tc-wleite**
This includes a basic (and hopefully fully functional) version of #569.
While implementing this I made minor changes in other UI related parts of the code, including:

- Create a HintTextField (from JTextField) that centralizes code related to showing a hint (with a different color) when a text field is empty (and it is not being edit). This is currently used in column manager frame and metadata panel.
- Make bookmark and column management windows only on top of IPED main window (not on top of other applications).
- In the gallery, use the JLabel to display "Thumbnail not supported" (instead of an Icon), to allow easier centering and line breaking. After #567, the text sometimes was not centered. 
- Change HexViewerPlus to only save configuration when user makes changes to default configuration.
- Minor fix to image viewer so adjusting the brightness for images with indexed colors or alpha channel works properly.

Tabs in metadata viewer, most viewers content (PDF, HTML), map and graph visualization were mostly kept with their original colors, because making them support color themes would require too many changes in their code and/or wouldn't make sense.

### CODE REVIEW

**Comment by lfcnassif** [comment]
Thank you @tc-wleite! Will try to take a look this week.

**Comment by tc-wleite** [comment]
> Thank you @tc-wleite! Will try to take a look this week.

Thanks! This is obviously low priority.
It is worth mentioning that changes in existing code were made in a way that there is no "hard dependency" on introduced themes, i.e. everything should work if theme management classes are removed.

**Comment by lfcnassif** [comment]
@tc-wleite I complemented this a little bit. I'vê made email header section of Eml and Msg viewers theme aware. But still need to handle Pst/Ost extracted emails, this would need a different approach, because in this case the email html preview is already built. I think this is possible to do, Will try today. Or we can revert Eml/MsgViewer changes. What do you think?

**Comment by tc-wleite** [comment]
> @tc-wleite I complemented this a little bit. 

Great! I saw the changes here! 
There were parts of the code that I was not very familiar with (e.g. graph tab), so I left for a future improvement as I mentioned in my initial post here, but you already covered most of them. 

> I've made email header section of Eml and Msg viewers theme aware. But still need to handle Pst/Ost extracted emails, this would need a different approach, because in this case the email html preview is already built. I think this is possible to do, Will try today. Or we can revert Eml/MsgViewer changes. What do you think?

When I was working on this, I talked a bit here with people that use dark mode "everywhere" (including the SO).
Even with that setup, when you open some windows/kind of files (e.g. PDF), it renders with the traditional light look.
So I think it is perfectly acceptable to not handle theme colors in all IPED viewers/formats.
On the other hand, as you already worked in EML/MSG, maybe it is worth trying to find a solution for PST, if it is simple enough.
If it isn't, I would (at least for) now revert (maybe leave commented, if possible) changes made for EML/MSG.

When I was working on this I found several option of dealing with HTML content. One of them was kind of dynamic, i.e. would work for "static" (already created) HTMLs, under certain conditions...
Give me an hour or so, I will try to find it here and see if can be applied to PSTs.

**Comment by lfcnassif** [comment]
> When I was working on this I found several option of dealing with HTML content. One of them was kind of dynamic, i.e. would work for "static" (already created) HTMLs, under certain conditions...

Ok, I was going to take a look later, but as you have already worked on it, I will do other thing. Just to be clear, my intent was just to make email headers (generated by the app) theme aware, not html content, it's ok to me to use default theme for content as you pointed.

**Comment by tc-wleite** [comment]
I think I found a simple solution...
The basic idea is to set a class in the generated HTML (ipedtheme), in places where is desired that theme colors are applied.
If there are no themes (or the HTML is externally opened), nothing is affected.
In the HTML Viewer this style is set (before loading a file).

I also set a different color for HTML links, as using the same as text makes a bit harder to spot links in message header.
IPED Help HTMLs were changed to use this solution too.

I applied (in separated commits) the same solution to MSG/EML just to keep things consistent, but feel free to revert.

**Comment by lfcnassif** [comment]
> I think I found a simple solution...
> The basic idea is to set a class in the generated HTML (ipedtheme), in places where is desired that theme colors are applied.
> If there are no themes (or the HTML is externally opened), nothing is affected.
> In the HTML Viewer this style is set (before loading a file).
> 
> I also set a different color for HTML links, as using the same as text makes a bit harder to spot links in message header.
> IPED Help HTMLs were changed to use this solution too.
> 
> I applied (in separated commits) the same solution to MSG/EML just to keep things consistent, but feel free to revert.

Great, @tc-wleite! Exactly what I was thinking about, thank you!

**Comment by lfcnassif** [comment]
> IPED Help HTMLs were changed to use this solution too.

So all improvements I thought about were done. I will do a bit more tests later and merge!

**Comment by tc-wleite** [comment]
> > IPED Help HTMLs were changed to use this solution too.
> 
> So all improvements I thought about were done. I will do a bit more tests later and merge!

I am running a few more tests here before pushing the changes.
Please take a closer look in these changes, as they affected parts of the code that I am not very familiar with (e.g. email parsers and viewers).

File: iped-viewers/iped-viewers-impl/src/main/java/dpf/sp/gpinf/indexer/ui/fileViewer/frames/HtmlViewer.java [hunk 1]
```diff
@@ -208,6 +208,11 @@ public void changed(ObservableValue<? extends Worker.State> ov, Worker.State old
                             doc = webEngine.getDocument();
 
                             if (doc != null) {
+                                String style = UiUtil.getUIHtmlStyle();
+                                if (style != null) {
+                                    webEngine.setUserStyleSheetLocation(style);
```
Comment (lfcnassif) [inline iped-viewers/iped-viewers-impl/src/main/java/dpf/sp/gpinf/indexer/ui/fileViewer/frames/HtmlViewer.java]: Nice method!

**Comment by lfcnassif** [comment]
Did a few more tests, no issues detected. Thank you @tc-wleite!

**Pull request merged**